src/storage/index.js

```diff
@@ -78,45 +92,57 @@ export function encryptContent(content: string | Buffer, options?: {privateKey?:
 export function decryptContent(content: string, options?: {privateKey?: ?string}) {
   const defaults = { privateKey: null }
   const opt = Object.assign({}, defaults, options)
-  if (! opt.privateKey) {
-    opt.privateKey = loadUserData().appPrivateKey
+  let privateKey = opt.privateKey
+  if (!privateKey) {
+    privateKey = loadUserData().appPrivateKey
   }
 
-  const cipherObject = JSON.parse(content)
-  return decryptECIES(opt.privateKey, cipherObject)
+  try {
+    const cipherObject = JSON.parse(content)
+    return decryptECIES(privateKey, cipherObject)
+  } catch (err) {
+    if (err instanceof SyntaxError) {
+      throw new Error('Failed to parse encrypted content JSON. The content may not ' +
+                      'be encrypted. If using getFile, try passing { decrypt: false }.')
+    } else {
+      throw err
+    }
+  }
 }
 
-/**
- * Retrieves the specified file from the app's data store.
- * @param {String} path - the path to the file to read
- * @param {Object} [options=null] - options object
- * @param {Boolean} [options.decrypt=true] - try to decrypt the data with the app private key
- * @param {String} options.username - the Blockstack ID to lookup for multi-player storage
- * @param {String} options.app - the app to lookup for multi-player storage -
- * defaults to current origin
- * @param {String} [options.zoneFileLookupURL=null] - The URL
- * to use for zonefile lookup. If falsey, this will use the
- * blockstack.js's getNameInfo function instead.
- * @returns {Promise} that resolves to the raw data in the file
- * or rejects with an error
+/* Get the gaia address used for servicing multiplayer reads for the given
+ * (username, app) pair.
+ * @private
  */
-export function getFile(path: string, options?: {decrypt?: boolean, username?: string, app?: string,
-  zoneFileLookupURL?: ?string}) {
-  const defaults = {
-    decrypt: true,
-    username: null,
-    app: window.location.origin,
-    zoneFileLookupURL: null
-  }
-
-  const opt = Object.assign({}, defaults, options)
+function getGaiaAddress(app: string, username: ?string, zoneFileLookupURL: ?string) {
```

Comment:
I thought about this part more.  The code assumes that the last "address-like" string in the Gaia hub URL is the base58check hash of the signer's public key.  I just want to be clear here--we're committing to requiring that the Gaia hub's URL contains the public key address?  If so, should we have an error path here that throws an exception if the URL does not have an address?